src/services/bunker_cases/abnormal_cl_rebase.py

```diff
@@ -21,187 +22,182 @@
                 {"msg": "Nearest and distant blocks are the same. Specific CL rebase will be calculated once"}
             )
-            specific_cl_rebase = self._calculate_cl_rebase_between(nearest_blockstamp, ref_blockstamp)
+            specific_cl_rebase = self._calculate_cl_rebase_between_blocks(nearest_blockstamp, blockstamp)
             logger.info({"msg": f"Specific CL rebase: {specific_cl_rebase} Gwei"})
             return specific_cl_rebase < 0
 
-        nearest_cl_rebase = self._calculate_cl_rebase_between(nearest_blockstamp, ref_blockstamp)
+        nearest_cl_rebase = self._calculate_cl_rebase_between_blocks(nearest_blockstamp, blockstamp)
         logger.info({"msg": f"Nearest specific CL rebase {nearest_cl_rebase} Gwei"})
         if nearest_cl_rebase < 0:
             return True
-        distant_cl_rebase = self._calculate_cl_rebase_between(distant_blockstamp, ref_blockstamp)
+
+        distant_cl_rebase = self._calculate_cl_rebase_between_blocks(distant_blockstamp, blockstamp)
         logger.info({"msg": f"Distant specific CL rebase {distant_cl_rebase} Gwei"})
         if distant_cl_rebase < 0:
             return True
+
         return False
 
-    def _calculate_cl_rebase_between(self, prev_blockstamp: ReferenceBlockStamp, ref_blockstamp: ReferenceBlockStamp) -> Gwei:
+    def _get_nearest_and_distant_blockstamps(
+        self, blockstamp: ReferenceBlockStamp
+    ) -> tuple[ReferenceBlockStamp, ReferenceBlockStamp]:
+        """Get nearest and distant blockstamps. Calculation including missed slots"""
+        nearest_slot = SlotNumber(
+            blockstamp.slot_number - self.b_conf.rebase_check_nearest_epoch_distance * self.c_conf.slots_per_epoch
+        )
+        distant_slot = SlotNumber(
+            blockstamp.slot_number - self.b_conf.rebase_check_distant_epoch_distance * self.c_conf.slots_per_epoch
+        )
+
+        AbnormalClRebase.validate_slot_distance(
+            distant_slot, nearest_slot, blockstamp.slot_number
+        )
+
+        nearest_blockstamp = self._get_ref_blockstamp(needed=nearest_slot, finalized=blockstamp.slot_number)
+
+        missed_slots = nearest_blockstamp.slot_number - nearest_slot
+
+        distant_blockstamp = self._get_ref_blockstamp(
+            needed=SlotNumber(distant_slot - missed_slots), finalized=blockstamp.slot_number
+        )
+
+        return nearest_blockstamp, distant_blockstamp
+
+    def _calculate_cl_rebase_between_blocks(
+        self, prev_blockstamp: ReferenceBlockStamp, curr_blockstamp: ReferenceBlockStamp
+    ) -> Gwei:
         """
-        Calculate CL rebase from prev_blockstamp to ref_blockstamp.
+        Calculate CL rebase from prev_blockstamp to curr_blockstamp.
         Skimmed validator rewards are sent to 0x01 withdrawal credentials address
         which in Lido case is WithdrawalVault contract. To account for all changes in validators' balances,
         we must account for WithdrawalVault balance changes (withdrawn events from it)
```

Comment:
`To account for all changes in validators' balances, we must account for WithdrawalVault balance changes (withdrawn events from it)`

it's not clear why. Maybe let's add smth like: between the prev_blockstamp and ref_blockstamp, an oracle report may occur, causing WithdrawalVault to be withdrawn. Check for these events is enough to account for all withdrawals since the protocol assumes that the vault can only be withdrawn at the time of the Oracle report

Reply:
fixed